dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Trn/CheckAnswers.cshtml.cs

```diff
@@ -31,8 +40,48 @@ public void OnGet()
     {
     }
 
-    public void OnPost()
+    public async Task<IActionResult> OnPost()
     {
+        var authenticationState = HttpContext.GetAuthenticationState();
+
+        var userId = Guid.NewGuid();
+        var user = new User()
+        {
+            CompletedTrnLookup = _clock.UtcNow,
+            Created = _clock.UtcNow,
+            DateOfBirth = authenticationState.DateOfBirth,
+            EmailAddress = authenticationState.EmailAddress!,
+            FirstName = string.IsNullOrEmpty(authenticationState.FirstName) ? authenticationState.OfficialFirstName! : authenticationState.FirstName,
+            LastName = string.IsNullOrEmpty(authenticationState.LastName) ? authenticationState.OfficialLastName! : authenticationState.LastName,
+            Updated = _clock.UtcNow,
+            UserId = userId,
+            UserType = UserType.Default,
+            Trn = authenticationState.StatedTrn,
```

Comment:
```suggestion
            Trn = authenticationState.Trn,
```

Reply:
Ah yes forgot to change after rebasing onto main